File: top_hot100/0094BinaryTreeInorderTraversal/inorder_traversal.go
package inordertraversal

// TreeNode 树的节点
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func inorderTraversal(root *TreeNode) []int {
	result := make([]int, 0)
	dfs(root, &result)
	return result
}

func dfs(root *TreeNode, result *[]int) {
	if root == nil {
		return
	}
	dfs(root.Left, result)
	*result = append(*result, root.Val)
	dfs(root.Right, result)
}

File: top_hot100/0094BinaryTreeInorderTraversal/inorder_traversal.py
# coding: utf-8

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def inorderTraversal(self, root):
        result = []
        self.dfs(root, result)
        return result
    
    def dfs(self, root, result):
        if root == None:
            return
        self.dfs(root.left, result)
        result.append(root.val)
        self.dfs(root.right, result)